telemetry_package/main.py

```diff
@@ -1,28 +1,74 @@
-import sys
+import json
 import rclpy
-from rclpy.node import Node
+import socket
+import os
+import select
 
+from rcl_interfaces.msg import Log
 from common_package_py.common_node import CommonNode
 
+DEFAULT_UNIX_SOCKET_PATH = "/tmp/thi_drone"
 
 class TelemetryNode(CommonNode):
-    """A class that represents a telemetry node.
-    """
 
-    def __init__(self, id: str):
-        """Creates a new TelemetryNode.
+    def __init__(self):
+        # The creation of the needed sockets takes place in the constructor
+        super().__init__('telemetry_node')
 
-        Args:
-            id (str): Unique node id
-        """
+        if os.path.exists(DEFAULT_UNIX_SOCKET_PATH): 
+            print('Socket already exists')
+            raise FileExistsError
+            
+        # Create server socket
+        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
-        super().__init__(id)
+        try:
+            # Bind server socket and listen for a connection. The node is blocked as long as a connection does not exist
+            self.server_sock.bind(DEFAULT_UNIX_SOCKET_PATH)
+            self.server_sock.listen(1)
+
+            # Getting the reference for the client socket
+            readable, _, _, = select.select([self.server_sock], [], [])
+            for sock in readable:
```

Comment:
What happens if there is nothing in readable?
You don't seem to capture this case.

Reply:
You could simply replace this whole for loop (including the if statement in it) with the following statement:
```suggestion
            if self.server_sock in readable:
```

Reply:
@maf0115, @Philipp66904 this code block is actually redundant. Just looked it up that listen is blocking if socket is not nonblocking. https://man7.org/linux/man-pages/man2/accept.2.html

We can actually throw it out.

Reply:
Just tried your change on my VM, it works. Will be applied and commited along other fixes